### PR TITLE
Add rotated to regular regrid recipe

### PR DESCRIPTION
**Pull request by bewithankit**
Added new recipe to regrid from a rotated latitude-longitude source domain to a regular latitude-longitude destination domain.

### CODE REVIEW

File: docs/source/recipes.rst [hunk 1]
```diff
@@ -408,4 +408,231 @@ In this recipe we will plot wind vectors, derived from northward and eastward wi
...
+          Bounds:Data(latitude(30), 2) = [[33.3448275862069, ..., 72.65517241379311]] degrees_north
+
+      Dimension coordinate: longitude
+          standard_name = 'longitude'
+          units = 'degrees_east'
+          Data(longitude(30)) = [-25.0, ..., 45.0] degrees_east
+          Bounds:units = 'degrees_east'
+          Bounds:Data(longitude(30), 2) = [[-26.20689655172414, ..., 46.20689655172414]] degrees_east
+
+9. Step 6 is similarly repeated for the regridded_data to plot the wind gust on a regular latitude-longitude domain:
```
Comment (davidhassell) [inline docs/source/recipes.rst:630]:
```suggestion
9. Step 6 is similarly repeated for the ``regridded_data`` to plot the wind gust on a regular latitude-longitude domain:
```

File: docs/source/recipes.rst [hunk 2]
```diff
@@ -408,4 +408,231 @@ In this recipe we will plot wind vectors, derived from northward and eastward wi
...
+
+7. To see the rotated pole data on the native grid, the above steps are repeated and projection is set to rotated in `cfplot.mapset <http://ajheaps.github.io/cf-plot/mapset.html>`_:
+
+   .. code-block:: python
+
+      >>> cfp.mapset(resolution='50m', proj='rotated')
+      >>> cfp.con(gust.subspace(T=test), lines=False)
+   .. figure:: images/windgust_rotated.png
+
+8. Create dimension coordinates for the destination grid with the latitude and longitude values for Europe using `~cf.DimensionCoordinate` class. `np.linspace <https://numpy.org/doc/stable/reference/generated/numpy.linspace.html>`_ generates evenly spaced values between the specified latitude and longitude range. Bounds of the target longitude and target latitude are created using `~cf.DimensionCoordinate.create_bounds` method. Spherical regridding is then performed on the gust variable by calling `~cf.Field.regrids` method and passing the target latitude and target longitude as arguments. The method also takes an argument 'linear' which specifies the type of regridding method to use. The desciption of the regridded_data is finally printed using the `~cf.Field.dump` method to show properties of all its constructs:
```
Comment (davidhassell) [inline docs/source/recipes.rst:560]:
```suggestion
8. Create dimension coordinates for the destination grid with the latitude and longitude values for Europe using `~cf.DimensionCoordinate` class. `np.linspace <https://numpy.org/doc/stable/reference/generated/numpy.linspace.html>`_ generates evenly spaced values between the specified latitude and longitude range. Bounds of the target longitude and target latitude are created using `~cf.DimensionCoordinate.create_bounds` method. Spherical regridding is then performed on the gust variable by calling the `~cf.Field.regrids` method and passing the target latitude and target longitude as arguments. The method also takes an argument ``'linear'`` which specifies the type of regridding method to use. The desciption of the regridded_data is finally printed using the `~cf.Field.dump` method to show properties of all its constructs:
```

File: docs/source/recipes.rst [hunk 3]
```diff
@@ -408,4 +408,231 @@ In this recipe we will plot wind vectors, derived from northward and eastward wi
...
+
+5. Create a new instance of the `cf.dt` class with a specified year, month, day, hour, minute, second and microsecond. Then store the result in the variable test:
+
+   .. code-block:: python
+
+      >>> test = cf.dt(2051, 4, 14, 1, 30, 0, 0)
+      >>> print(test)
+      2051-04-14 01:30:00
+
+6. Plot the wind gust by creating a (`~cf.Field.subspace`) for the specified variable 'test' using `cfplot.con <http://ajheaps.github.io/cf-plot/con.html>`_. Here `cfplot.mapset <http://ajheaps.github.io/cf-plot/mapset.html>`_ is used to set the mapping parameters like setting the map resolution to 50m:
```
Comment (davidhassell) [inline docs/source/recipes.rst:544]: Are the brackets around "subspace" needed?
```suggestion
6. Plot the wind gust by creating a `~cf.Field.subspace` for the specified variable ``test`` using `cfplot.con <http://ajheaps.github.io/cf-plot/con.html>`_. Here `cfplot.mapset <http://ajheaps.github.io/cf-plot/mapset.html>`_ is used to set the mapping parameters like setting the map resolution to 50m:
```

File: docs/source/recipes.rst [hunk 4]
```diff
@@ -408,4 +408,231 @@ In this recipe we will plot wind vectors, derived from northward and eastward wi
...
+          Coordinate conversion:grid_mapping_name = rotated_latitude_longitude
+          Coordinate conversion:grid_north_pole_latitude = 39.25
+          Coordinate conversion:grid_north_pole_longitude = 198.0
+          Dimension Coordinate: grid_longitude
+          Dimension Coordinate: grid_latitude
+          Auxiliary Coordinate: longitude
+          Auxiliary Coordinate: latitude
+
+
+4. Access the time coordinate of the gust field using `~cf.Field.coordinate` method, retrieve the datetime values of the time coordinate using `~cf.DimensionCoordinate.datetime_array` and print the resulting datetime array:
```
Comment (davidhassell) [inline docs/source/recipes.rst:522]:
```suggestion
4. Access the time coordinate of the gust field using `~cf.Field.coordinate` method and retrieve the datetime values of the time coordinate using `~cf.DimensionCoordinate.datetime_array`:
```

**Review by davidhassell — CHANGES_REQUESTED** [review]
Hi Ankit. Looks good. A few minor suggetions. Please merge when you've decided what to do with them. Thanks

**Comment by bewithankit** [comment]
Thank you for the suggestions David!

**Comment by bewithankit** [comment]
Merging now!

**Pull request merged**